hls4ml/templates/vivado/nnet_utils/nnet_conv1d_latency.h

```diff
@@ -84,5 +84,356 @@ void conv_1d_latency_cl(data_T data[CONFIG_T::in_width * CONFIG_T::n_chan],
+
+    data_T data_tmp[CONFIG_T::reuse_factor][CONFIG_T::in_width * CONFIG_T::n_chan / CONFIG_T::reuse_factor];
+    #pragma HLS ARRAY_PARTITION variable=data_tmp complete dim=0
+    res_T res_tmp[CONFIG_T::reuse_factor][CONFIG_T::out_width * CONFIG_T::n_filt / CONFIG_T::reuse_factor];
+    #pragma HLS ARRAY_PARTITION variable=res_tmp complete dim=0
+
+RFInputLoop:
+    for (int jj = 0; jj < CONFIG_T::reuse_factor; jj++) {
+    #pragma HLS UNROLL
+    InnerInputLoop:
+        for (int ii = 0; ii < CONFIG_T::in_width * CONFIG_T::n_chan / CONFIG_T::reuse_factor; ii++) {
+            #pragma HLS UNROLL
+            data_tmp[jj][ii] = data[jj * CONFIG_T::in_width * CONFIG_T::n_chan / CONFIG_T::reuse_factor + ii];
+        }
+    }
+
+    pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[0], res_tmp[0], weights, biases);
+    pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[1], res_tmp[1], weights, biases);
+    if (CONFIG_T::reuse_factor > 2)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[2], res_tmp[2], weights, biases);
+    if (CONFIG_T::reuse_factor > 3)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[3], res_tmp[3], weights, biases);
+    if (CONFIG_T::reuse_factor > 4)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[4], res_tmp[4], weights, biases);
+    if (CONFIG_T::reuse_factor > 5)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[5], res_tmp[5], weights, biases);
+    if (CONFIG_T::reuse_factor > 6)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[6], res_tmp[6], weights, biases);
+    if (CONFIG_T::reuse_factor > 7)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[7], res_tmp[7], weights, biases);
+    if (CONFIG_T::reuse_factor > 8)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[8], res_tmp[8], weights, biases);
+    if (CONFIG_T::reuse_factor > 9)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[9], res_tmp[9], weights, biases);
+    if (CONFIG_T::reuse_factor > 10)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[10], res_tmp[10], weights, biases);
+    if (CONFIG_T::reuse_factor > 11)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[11], res_tmp[11], weights, biases);
+    if (CONFIG_T::reuse_factor > 12)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[12], res_tmp[12], weights, biases);
+    if (CONFIG_T::reuse_factor > 13)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[13], res_tmp[13], weights, biases);
+    if (CONFIG_T::reuse_factor > 14)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[14], res_tmp[14], weights, biases);
+    if (CONFIG_T::reuse_factor > 15)
+        pointwise_conv_1d_latency_cl<data_T, res_T, CONFIG_T>(data_tmp[15], res_tmp[15], weights, biases);
+    if (CONFIG_T::reuse_factor > 16)
```

Comment:
I'm wondering if there is a better way to do this ...

Reply:
yes, i think we can use the code-generation machinery like this:
https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/backends/fpga/passes/codegen.py
https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/backends/fpga/fpga_backend.py#L671-L731

Reply:
@vloncar @Duchstf I started this branch to use the code generation machinery: https://github.com/jmduarte/hls4ml/pull/20

direct diff: https://github.com/fastmachinelearning/hls4ml/compare/main...jmduarte:split_pointwise_conv_by_rf_codegen

is this better than the current approach?